code/modules/projectiles/gun.dm

```diff
@@ -75,6 +75,16 @@
 
 	var/spread = 0 //Spread induced by the gun itself.
 	var/randomspread = 1 //Set to 0 for shotguns. This is used for weapons that don't fire all their bullets at once.
+	var/dual_wield_spread = 24 //additional spread when dual wielding
+
+	///Screen shake when the weapon is fired
+	var/recoil = 0
+	///a multiplier of the duration the recoil takes to go back to normal view, this is (recoil*recoil_backtime_multiplier)+1
+	var/recoil_backtime_multiplier = 1.5
+	///this is how much deviation the gun recoil can have, recoil pushes the screen towards the reverse angle you shot + some deviation which this is the max.
+	var/recoil_deviation = 20
+	///Used if the guns recoil is lower then the min, it clamps the highest recoil
+	var/min_recoil = 0
```

Comment:
I feel this var should be renamed to `cosmetic_recoil`
At the moment it's a little misleading, as it's not just the lower end of the clamp (it's affected by the value of the preference).

Alternatively you can just update the autodoc and say user error if someone messes it up
```
/// Used as the min value when calculating recoil 
/// Affected by a player's min_recoil_multiplier preference, so keep in mind it can ultimately be 0 regardless
/// Often utilized as a "purely visual" form of recoil (as it can be disabled)  
```